arrows/core/create_detection_grid.cxx

```diff
@@ -0,0 +1,180 @@
+
+  /// Logger handle
+  vital::logger_handle_t m_logger;
+};
+
+/// Constructor
+create_detection_grid
+::create_detection_grid()
+  : d_( new priv )
+{
+}
+
+
+/// Destructor
+create_detection_grid
+::~create_detection_grid() noexcept
+{
+}
+
+
+/// Get this alg's \link vital::config_block configuration block \endlink
+vital::config_block_sptr
+create_detection_grid
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "detection_width", d_->width,
+    "Width of each detection in the output grid." );
+  config->set_value( "detection_height", d_->height,
+    "Height of each detection in the output grid." );
+
+  return config;
+}
+
+
+/// Set this algo's properties via a config block
+void
+create_detection_grid
+::set_configuration( vital::config_block_sptr in_config )
+{
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  d_->width = config->get_value<double>( "detection_width" );
+  d_->height = config->get_value<double>( "detection_height" );
+}
+
+
+bool
+create_detection_grid
+::check_configuration(vital::config_block_sptr config) const
+{
+  if(config->get_value<double>("detection_width") <= 0 || config->get_value<double>("detection_width") <= 0)
+  {
+    LOG_ERROR(d_->m_logger, "Detection width and height must be positive values");
+    return false;
+  }
+  return true;
+}
+
+
+vital::detected_object_set_sptr
+create_detection_grid::
+detect( vital::image_container_sptr image_data) const
+{
+  vital::detected_object_set_sptr grid(new vital::detected_object_set());
+  size_t img_width = image_data->width();
+  size_t img_height = image_data->height();
+
+  if(d_->width > img_width || d_->height > img_height)
+  {
+    throw vital::algorithm_configuration_exception(
+      type_name(), impl_name(), "Detection width and height must be no more than image width and height");
+  }
+
+  // Get any non-overlapping grid spaces
+  // Note that the last column and row are missing here
+  for(int i = 0; i + d_->width < img_width ; i+=d_->width)
+  {
+    for(int j = 0; j + d_->height < img_height ; j+=d_->height)
+    {
+      vital::bounding_box<double> bbox(i, j, i+d_->width-1, j+d_->height-1);
+      vital::detected_object_sptr det_obj(new vital::detected_object(bbox));
+      grid->add(det_obj);
+    }
+  }
+
+  // Now get the bottom row
+  for (int i = 0; i + d_->width < img_width ; i+=d_->width)
+  {
+    vital::bounding_box<double> bbox(i, img_height - d_->height, i+d_->width-1, img_height-1);
+    vital::detected_object_sptr det_obj(new vital::detected_object(bbox));
+    grid->add(det_obj);
+  }
+
+  // Now get the bottom column
+  for (int j = 0; j + d_->height < img_height ; j+=d_->height)
+  {
+    vital::bounding_box<double> bbox(img_width - d_->width, j , img_width-1, j+d_->height-1);
+    vital::detected_object_sptr det_obj(new vital::detected_object(bbox));
```

Comment:
What if `img_width - d_->width ` is zero or negative?

Reply:
Again, there's a check for that above.